### PR TITLE
Fixes to sso config calls

### PR DESCRIPTION
**Pull request by aviadl**
related to https://github.com/descope/etc/issues/5353
+ tests

### CODE REVIEW

File: lib/management/sso.ts [hunk 1]
```diff
@@ -121,7 +128,27 @@ const withSSOSettings = (sdk: CoreSdk, managementKey?: string) => ({
...
+        if (readySettings.oidc) {
+          readySettings.oidc = {
+            ...readySettings.oidc,
+            attributeMapping: readySettings.oidc.userAttrMapping,
+          };
+          delete readySettings.oidc.userAttrMapping;
+        }
+        const mappings: RoleMappings = [];
+        readySettings.saml?.groupsMapping?.forEach((gm: any) => {
+          const rm = gm;
```
Comment (asafshen) [inline lib/management/sso.ts:142]: naming feels like go :)

File: lib/management/sso.ts [hunk 2]
```diff
@@ -121,7 +128,27 @@ const withSSOSettings = (sdk: CoreSdk, managementKey?: string) => ({
...
+          rm.roleName = rm.role.name;
+          delete rm.role;
+          mappings.push(rm);
+        });
+        if (readySettings.saml?.groupsMapping) {
+          readySettings.saml.groupsMapping = mappings;
+        }
+        return data;
```
Comment (asafshen) [inline lib/management/sso.ts:150]: not critical, but the condition is a bit mouthful
we can do it with map and avoid it
```suggestion
        readySettings.saml?.groupsMapping?.forEach((gm: any) => {
          const rm = gm;
          rm.roleName = rm.role.name;
          delete rm.role;
          mappings.push(rm);
        });
        readySettings.saml?.groupsMapping = readySettings.saml?.groupsMapping.map((gm: any) => {
          const rm = gm;
          rm.roleName = rm.role.name;
          delete rm.role;
          return rm;
        });
        return data;
```

really not critical

Comment (asafshen) [inline lib/management/sso.ts:150]: pointers and all, but shouldn't we return `readySettings` here?

**Review by asafshen — DISMISSED** [review]

File: lib/management/sso.ts (hunk 2 above)
Comment (aviadl) [reply lib/management/sso.ts:150]: done

Comment (aviadl) [reply lib/management/sso.ts]: done

**Review by asafshen — APPROVED** [review]

**Pull request merged**